tests/Unit/DonationForms/Rules/TestHoneyPotRule.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Give\Tests\Unit\DonationForms\Actions;
+
+use Give\DonationForms\Rules\HoneyPotRule;
+use Give\Tests\TestCase;
+use Give\Tests\TestTraits\RefreshDatabase;
+use Give\Tests\Unit\DonationForms\TestTraits\HasValidationRules;
+
+/**
+ * @unreleased
+ */
+class TestHoneyPotRule extends TestCase
+{
+    use RefreshDatabase;
+    use HasValidationRules;
+
+    /**
+     * @unreleased
+     * @dataProvider honeyPotProvider
+     */
+    public function testHoneyPotRule($value, bool $shouldBeValid): void
+    {
+        $rule = new HoneyPotRule();
+
+        if ($shouldBeValid) {
```

Comment:
Would it be better to have a `validProvider` and a `invalidProvider` with separate tests, so that the test names are more precise (`testHoneyPotValidatesEmpty`, `testHoneypotInvalidatesSubmittedValues`) and the added logic within each test is reduced?

Reply:
Ah, ok. That makes sense. In that case, let's not fiddle with it at this time.